spacy_llm/models/hf/falcon.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Callable, Dict, Iterable, Optional, Tuple
+
+from spacy.util import SimpleFrozenDict
+
+from ...compat import Literal, transformers
+from ...registry.util import registry
+from .base import HuggingFace
+
+
+class Falcon(HuggingFace):
+    def __init__(
+        self,
+        variant: str,
+        config_init: Optional[Dict[str, Any]],
+        config_run: Optional[Dict[str, Any]],
+    ):
+        self._tokenizer: Optional["transformers.AutoTokenizer"] = None
+        self._device: Optional[str] = None
+        super().__init__(
```

Comment:
Is there a reason why this isn't first statement in the ctor?

Reply:
The parent constructor calls the `init_model()` method, which in this subclass accesses `self._tokenizer` and `self._device`.